test/helpers/ssh_command.go

```diff
@@ -117,62 +117,78 @@ func ImportSSHconfig(config []byte) (SSHConfigs, error) {
 	return result, nil
 }
 
-//RunCommand runs a SSHCommand using SSHClient client. It will return the
-//stdout and a error.
-//Error will happen when a session can't be initialized correctly
-func (client *SSHClient) RunCommand(cmd *SSHCommand) ([]byte, error) {
-	var (
-		session *ssh.Session
-		err     error
-	)
-
-	if session, err = client.newSession(); err != nil {
-		return nil, err
-	}
-	defer session.Close()
+// copyWait runs an instance of io.Copy() in a goroutine, and returns a channel
+// to receive the error result.
+func copyWait(dst io.Writer, src io.Reader) chan error {
+	c := make(chan error)
+	go func() {
+		_, err := io.Copy(dst, src)
+		c <- err
```

Comment:
You should close the channel here.

Reply:
I discussed this a bit with @ianvernon as well, but even Ian Lance Taylor states that channels needn't be closed if you don't need to know that there will be no further messages --- the channel will be garbage-collected:

> The close function should only be used on a channel used to send data,
after all data has been sent.  It does not make sense to send more data
after the channel has been closed; if all data has not been sent, the
channel should not be closed.  Similarly, it does not make sense to
close the channel twice.
Note that it is only necessary to close a channel if the receiver is
looking for a close.  Closing the channel is a control signal on the
channel indicating that no more data follows.

https://groups.google.com/forum/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ